slsDetectorSoftware/src/Detector.cpp

```diff
@@ -1612,24 +1612,27 @@ void Detector::setPatternBitMask(uint64_t mask, Positions pos) {
 
 // Moench
 
-Result<std::string> Detector::getAdditionalJsonHeader(Positions pos) const {
+Result<std::map<std::string, std::string>> Detector::getAdditionalJsonHeader(Positions pos) const {
     return pimpl->Parallel(&Module::getAdditionalJsonHeader, pos);
 }
 
-void Detector::setAdditionalJsonHeader(const std::string &jsonheader,
+void Detector::setAdditionalJsonHeader(const std::map<std::string, std::string> jsonHeader,
                                        Positions pos) {
-    pimpl->Parallel(&Module::setAdditionalJsonHeader, pos, jsonheader);
+    pimpl->Parallel(&Module::setAdditionalJsonHeader, pos, jsonHeader);
 }
 
 Result<std::string> Detector::getAdditionalJsonParameter(const std::string &key,
                                                          Positions pos) const {
     return pimpl->Parallel(&Module::getAdditionalJsonParameter, pos, key);
 }
 
-void Detector::setAdditionalJsonParameter(const std::string &key,
-                                          const std::string &value,
+void Detector::setAdditionalJsonParameter(const std::map<std::string, std::string> para,
```

Comment:
Do we need a map here, isn't it enough to take const std::string& key, const std::string& parameter? If we just set one parameter?

Reply:
I agree. It came from wanting to allow multiple json parameters, but removing this later. Will fix this.

Reply:
done